app/helpers/sources_helper.rb

```diff
@@ -13,4 +13,35 @@ def render_media_asset
       render partial: 'video'
     end
   end
+
+  ##
+  # Render a back link to the referring page if the referring page is a guide or
+  # set.  Links from guides or sets to sources may be created by admins, so the
+  # applicaiton cannot expect that back URI's can be included in params.
+  # Therefore, it is necessary to parse the referring URI to see if it is a
+  # guide or set, and render a back link accordingly.
+  #
+  # @return [String] HTML back link
+  def render_back_link
+    return unless request.referer.present?
+    uri = URI(request.referer)
+    # return if referrer is an offsite link
+    return unless uri.host == request.host
```

Comment:
Good question. I [amended the code](https://github.com/dpla/primary-source-sets/pull/120/files#diff-89d53548c3df2e0e1fd19f6e08527037R33) to return with no action if the URL does not have the correct relative root (ie. `/primary-source-sets`).  I also added a [test](https://github.com/dpla/primary-source-sets/pull/120/files#diff-08b75f9ca481ff07212281153c81226cR57) for scenario you described above.